ds3/networking/httpRequestBuilder.go

```diff
@@ -104,6 +104,13 @@ func (builder *HttpRequestBuilder) Build(conn *ConnectionInfo) (*http.Request, e
 
     if builder.contentLength != nil {
         httpRequest.ContentLength = *builder.contentLength
+
+        // Special casing for content length == 0.  Go won't include the content length header
+        // if the length is 0, but BlackPearl needs the content length header to be there and be 0
+        // to create a folder.
+        if *builder.contentLength == 0 {
+            httpRequest.Body = http.NoBody
```

Comment:
by setting an empty body you enforce the add of the length header? and automatically set it to 0?

Reply:
Sort of.  httpRequest.ContentLength = *builder.contentLength sets the content-length header.
See https://golang.org/doc/go1.8#net_http.  Setting the body to nil or http.NoBody, which is what the docs recommend, prevents the http library from removing the content-length 0 header.